explainaboard/processors/extractive_qa.py

```diff
@@ -22,77 +28,75 @@ class QAExtractiveProcessor(Processor):
+                    if isinstance(x["answers"]["text"], list)
+                    else x['answers']['text'],
+                    side='target',
                 ),
-                "answer_length": feature.Value(
-                    dtype="float",
-                    description="answer length",
-                    is_bucket=True,
-                    bucket_info=feature.BucketInfo(
-                        method="bucket_attribute_specified_bucket_value",
-                        number=4,
-                        setting=(),
-                    ),
+            ),
+            "num_oov": feature.Value(
+                dtype="float",
+                description="the number of out-of-vocabulary words in the context",
+                require_training_set=True,
+                func=lambda info, x, c, stat: feat_num_oov(
+                    info, x['context'], stat['source_vocab']
                 ),
-                "num_oov": feature.Value(
-                    dtype="float",
-                    description="the number of out-of-vocabulary words",
-                    is_bucket=True,
-                    bucket_info=feature.BucketInfo(
-                        method="bucket_attribute_specified_bucket_value",
-                        number=4,
-                        setting=(),
-                    ),
-                    require_training_set=True,
+            ),
+            "fre_rank": feature.Value(
+                dtype="float",
+                description=(
+                    "average rank of context words based on training set freq"
                 ),
-                "fre_rank": feature.Value(
-                    dtype="float",
-                    description=(
-                        "the average rank of each word based on its frequency in "
-                        "training set"
-                    ),
-                    is_bucket=True,
-                    bucket_info=feature.BucketInfo(
-                        method="bucket_attribute_specified_bucket_value",
-                        number=4,
-                        setting=(),
-                    ),
-                    require_training_set=True,
+                require_training_set=True,
+                func=lambda info, x, c, stat: feat_freq_rank(
+                    info, x['context'], stat['source_vocab_rank']
                 ),
-            }
-        )
+            ),
+        }
+        continuous_features = [
+            k for k, v in features.items() if ('float' in unwrap(v.dtype))
+        ]
+        analyses: list[BucketAnalysis] = [
+            BucketAnalysis(
+                description=features[x].description, feature=x, method="continuous"
+            )
+            for x in continuous_features
+        ]
+
+        return [
+            AnalysisLevel(
+                name='example',
+                features=features,
+                metric_configs=self.default_metrics(),
+                analyses=cast(List[Analysis], analyses),
```

Comment:
Any reason to use `List`?
(ditto for others)

Reply:
This was due to a peculiarity of Python 3.8, but this code has been removed so it's no long relevant.